packages/components/src/color-indicator/index.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import type { WordPressComponentProps } from '../ui/context';
+import type { ColorIndicatorProps } from './types';
+
+/**
+ * ColorIndicator is a React component that renders a specific color in a
+ * circle. It's often used to summarize a collection of used colors in a child
+ * component.
+ *
+ * ```jsx
+ * import { ColorIndicator } from '@wordpress/components';
+ *
+ * const MyColorIndicator = () => <ColorIndicator colorValue="#0073aa" />;
+ * ```
+ */
+export function ColorIndicator(
+	// ref is omitted until we have `WordPressComponentPropsWithoutRef` or add
+	// ref forwarding to ColorIndicator.
+	props: Omit<
+		WordPressComponentProps< ColorIndicatorProps, 'span', false >,
+		'ref'
+	>
+) {
```

Comment:
Should we `forwardRef` here as well? cc @mirka

Reply:
Let me know and I'll update the PR / do a follow-up PR.

Reply:
Yep, let's forward! (In this PR or follow-up, whichever is most convenient to you)

Reply:
We've just updated the [Migration guidelines](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript) to include a point re. this  (point `7 / vii`):

Example snippet:

```tsx
// With `forwardRef`
import type { ForwardedRef } from 'react';
import { forwardRef } from '@wordpress/element';
import type { WordPressComponentProps } from '../ui/context';
import type { ComponentOwnProps } from './types';

function UnforwardedMyComponent(
	props:  WordPressComponentProps< ComponentOwnProps, 'div', true >,
	forwardedRef: ForwardedRef< any >
) { /* ... */ }


/**
 * MyComponent JSDoc
 */
export const MyComponent = forwardRef( UnforwardedMyComponent );

export default MyComponent;
```

Reply:
Been away for a few days. Will need to either make a follow-up on update this next week.